daemon/src/main/java/org/sonarlint/daemon/services/StandaloneSonarLintImpl.java

```diff
@@ -19,37 +19,159 @@
+      this.userObject = userObject;
+    }
+
+    @Override
+    public Path getPath() {
+      return path;
+    }
+
+    @Override
+    public boolean isTest() {
+      return isTest;
+    }
+
+    @Override
+    public Charset getCharset() {
+      return charset;
+    }
+
+    @Override
+    public <G> G getClientObject() {
+      return (G) userObject;
+    }
+  }
+
+  static class ProxyIssueListener implements IssueListener {
+    private final StreamObserver<Issue> observer;
+
+    protected ProxyIssueListener(StreamObserver<Issue> observer) {
+      this.observer = observer;
+    }
+
+    @Override
+    public void handle(org.sonarsource.sonarlint.core.client.api.common.analysis.Issue issue) {
+      Severity severity;
+      ClientInputFile inputFile = issue.getInputFile();
+
+      switch (issue.getSeverity()) {
+        case "MINOR":
+          severity = Severity.MINOR;
+          break;
+        case "BLOCKER":
+          severity = Severity.BLOCKER;
+          break;
+        case "INFO":
+          severity = Severity.INFO;
+          break;
+        case "CRITICAL":
+          severity = Severity.CRITICAL;
+          break;
+        case "MAJOR":
+        default:
+          severity = Severity.MAJOR;
+          break;
+      }
+
+      Issue.Builder builder = Issue.newBuilder();
+      builder.setRuleKey(issue.getRuleKey())
+        .setRuleName(issue.getRuleName())
+        .setMessage(issue.getMessage())
+        .setSeverity(severity)
+        .setStartLine(issue.getStartLine() != null ? issue.getStartLine() : 0)
+        .setStartLineOffset(issue.getStartLineOffset() != null ? issue.getStartLineOffset() : 0)
+        .setEndLine(issue.getEndLine() != null ? issue.getEndLine() : 0)
+        .setEndLineOffset(issue.getEndLineOffset() != null ? issue.getEndLineOffset() : 0);
+
+      if (inputFile != null) {
+        builder.setFilePath(inputFile.getPath().toString())
+          .setUserObject((String) inputFile.getClientObject());
+      }
+
+      observer.onNext(builder.build());
+    }
+  }
+
+  static class ProxyLogOutput implements LogOutput {
+    private StreamObserver<LogEvent> response;
+
+    protected void setObserver(StreamObserver<LogEvent> response) {
+      if (this.response != null) {
+        this.response.onCompleted();
+      }
+      this.response = response;
+    }
+
+    @Override
+    public synchronized void log(String formattedMessage, Level level) {
+      if (level == Level.ERROR) {
+        System.err.println(formattedMessage);
+      } else {
+        System.out.println(formattedMessage);
+      }
+
+      if (response != null) {
+        LogEvent log = LogEvent.newBuilder()
+          .setLevel(level.name())
+          .setLog(formattedMessage)
+          .setIsDebug(level == Level.DEBUG || level == Level.TRACE)
+          .build();
+        try {
+          response.onNext(log);
+        } catch (StatusRuntimeException e) {
+          System.out.println("Log stream closed: " + e.getMessage());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS106)